src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -39,28 +39,17 @@ static bool is_supported_statement(const ast::Statement& statement) {
            statement.is_if_statement() || statement.is_while_statement();
 }
 
-bool CodegenLLVMVisitor::check_array_bounds(const ast::IndexedName& node, unsigned index) {
```

Comment:
Yes, basically we have 3 cases:
1) Indexing with a single integer `x[1]`. In this case it is easy to check the bounds.
2) Indexing by expression `x[2 - 2]`. In this case we need to actually compute the values.
3) Indexing with names `x[id]`. In this case the analysis is not trivial as we have to track at the value changes, etc.

So i decided just to remove the check for now. It can be added later as a separate patch but right now not strictly necessary.